src/main/scala/com.snowplowanalytics/forex/Forex.scala

```diff
@@ -153,210 +131,156 @@ case class ForexLookupTo(conversionAmount: Double, fromCurr: String, fx: Forex)
  * @param conversionAmount - The amount of money to be converted, it is set to 1 for lookup operation
  * @param fromCurr - The source currency
  * @param toCurr   - The target currency
- * @param fx       - Forex object
+ * @param config   - Forex config
+ * @param client   - Passed down client that does actual work
  */
-case class ForexLookupWhen(conversionAmount: Double, fromCurr: String, toCurr: String, fx: Forex) {
+case class ForexLookupWhen[F[_]: Sync](conversionAmount: Double,
```

Comment:
Wondering if can factor our `Sync` from here as Andrzej did everywhere. Also, wondering if it is possible to do in our LRUMap. I believe typeclass constraints should not be part of data type definitions, but functions' instead.